### PR TITLE
Plumed: Add a new ArrayFire variant.

### PR DESCRIPTION
**Pull request by RemiLacroix-IDRIS**
Note that this depends on PR #20615.

### CODE REVIEW

File: var/spack/repos/builtin/packages/plumed/package.py [hunk 1]
```diff
@@ -57,6 +57,9 @@ class Plumed(AutotoolsPackage):
     variant('shared', default=True, description='Builds shared libraries')
     variant('mpi', default=True, description='Activates MPI support')
     variant('gsl', default=True, description='Activates GSL support')
+    variant('arrayfire', default='disable',
+            values=('disable', 'cpu', 'cuda', 'opencl'),
```
Comment (RemiLacroix-IDRIS) [inline var/spack/repos/builtin/packages/plumed/package.py]: Note that I am not sure about `disable` here, I think it should be `disabled` but maybe someone can come up with something better.

Comment (alalazo) [reply var/spack/repos/builtin/packages/plumed/package.py]: We usually use the string `none` if there is no support.

Comment (RemiLacroix-IDRIS) [reply var/spack/repos/builtin/packages/plumed/package.py]: Yeah, not sure why I didn't like `none` at first... I've repushed.

File: var/spack/repos/builtin/packages/plumed/package.py [hunk 2]
```diff
@@ -159,18 +165,31 @@ def configure_args(self):
...
-            blas_libs = self.spec['blas'].libs
-            configure_opts.append('LDFLAGS={0}'.format(
+        if '+gsl' in spec:
+            gsl_libs = spec['gsl'].libs
+            blas_libs = spec['blas'].libs
+            extra_libs.append(
                 (gsl_libs + blas_libs).ld_flags
+            )
+        # Set flags to help with ArrayFire
+        if 'arrayfire=disable' not in spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/plumed/package.py]:
```suggestion
        if 'arrayfire=none' not in spec:
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**